server/app.js

```diff
@@ -43,12 +43,11 @@ app.use(express.json());
 
 app.use('/', router);
 
-app.use(errorHandler);
-
 const { version } = require('../package.json');
-
 app.get('*', function (req, res) {
   res.status(200).send(version);
 });
 
+app.use(errorHandler);
```

Comment:
Why do you move the errorHandler below the star route? I thought it's problematic?

Reply:
I'm not aware of it being problematic.  I switched the order because it seemed like the version was what we wanted to respond with for any route that didn't match a valid one, and the error handler was blocking that by coming first.  But if there is an error it will bypass all the routes and go straight to the error handler.

Why is this order problematic?

Reply:
I remember the order is by intentional, I mean, we must put it before the '*' router, or we can not let the errorhandler work, because the '*' will block all the request and response the 'vx.x.x', so it's impossible to make the express handle chain to go to the errorhandler.
It's fine to put the errorhandle before the '*', because it will check the 'error' chain, if there isn't any error, then passby the chain to '*',